dignified-js.md

```diff
@@ -0,0 +1,206 @@
+
+#### b) Test
+
+All the code we write is expected to run in browser as well as in Node.js.
+We restrict ourselves to Node.js 4 and 5 and the latest stable releases
+of Chrome, Firefox, Safari and Edge at the moment.  Even this is quite a
+large target to hit, so our tests need to reflect that. Testing for Node.js
+is straightforward for the most part. The browser though wants a
+bit more love. So we are using [karma](http://karma-runner.github.io)
+to automate the test execution in the browser, as we are familiar with
+it (given I'm the maintainer of it).
+
+To reduce friction and overhead, we share the test
+framework [mocha](http://mochajs.org/) and the assertion library [chai](http://chaijs.com/) between the browser and Node.js.
+
+#### c) Build
+
+There are a lot of build systems out there, and I have used my
+fair share of them in the past ([r.js](http://requirejs.org/), [rollup](http://rollupjs.org/), [browserify](http://browserify.org/), [webpack](http://webpack.github.io/)). They all have their strength and weaknesses and at the
+end of the day it matters that the tool does what you want it to do. For
+that reason we settled on using webpack as it gives us a large control over
+every detail when bundling and we feel quite comfortable with it at this
+point.
+
+As we are using some features of ES2015 we also need something to ensure
+our code runs on the platforms that do not have full support yet. For this
+[babel](http://babeljs.io/) is the tool of our choice.
+
+When others consume our code though, we don’t want to enforce these choices
+on them. So that’s why our build process in dignified.js creates multiple
+versions for (hopefully) everyone to use.
+
+- __Raw ES2015 version__, ready to be consumed by platforms that understand Node.js based require and most of ES2015.
+- __Raw ES5 version__, ready to be consumed by platforms that understand Node.js based require and ES5.
+- __Concatenated ES5 version__, ready to be consumed by browsers through a script tag, where size does not matter.
+- __Concatenated and minified ES5 version__, ready to be consumed by browsers through a script tag, where size matters.
+
+#### d) Release
+
+This is actually quite simple, we have a [gulp](http://gulpjs.com/) task
+that runs everything needed for a release so we don't miss
+anything of the above mentioned things. It will:
+
+1. Run linting
+2. Run the tests
+3. Build all versions
+4. Bump the version in package.json
+5. Commit the version bump
+6. Create a git tag
+7. Push this to github
+8. Publish to npm
+
+### Day to day operations...
+
+#### ...for maintainers
+
+There are a couple of binaries that dignified.js provides for you to use
+
+```sh
+$ dignified-lint$ dignified-test
+$ dignified-test browser
+$ dignified-test node
+$ dignified-build
+$ dignified release major
+$ dignified release minor
+$ dignified release
```

Comment:
Probably should be dignified-release for consistency... Also that is what it says in the scripts section below.